### PR TITLE
Fix navigation tree not showing up

### LINKED ISSUE #256: Can't see nor edit navigation tree
Prerequisites
* [x] Are you running the latest version?
* [x] Are you able to consistently reproduce the issue?
* [x] Did you search the issue queue for existing issue? [Search issues](https://github.com/reactioncommerce/reaction/issues)

## Issue Description
After installing a brand new Reaction Commerce environment with the latest `reaction-admin`, the "Navigation" page doesn't show a navigation tree. Navigation items are showing up fine on the left, but the right is essentially a blank area. It is thus impossible to modify anything navigation related as of now.

Screen Shot 2020-04-13 at 16 12 14

## Steps to Reproduce
Please provide starting context, i.e. logged in as a user, configure a particular payment method.
1. Log in a user
2. Click the Navigation link in the sidebar
3. Notice that your navigation tree isn't showing up

### Versions
`3.0.0-beta.6`

### PR DESCRIPTION
**Pull request by loan-laux**
Signed-off-by: Loan Laux <loan@outgrow.io>

Resolves #256 
Impact: **critical**
Type: **bugfix**

## Issue
The Navigation page wasn't working because of the navigation tree not showing up.

## Solution
Update `react-dnd` and `react-dnd-html5-backend` to 9.5.1, and move `DndProvider` up the component tree (at `<App />` level) to avoid conflict with `react-router`.

## Breaking changes
None.

## Testing
1. Visit the Navigation page.
2. See the navigation tree.
3. Try moving stuff around. It should work.

### CODE REVIEW

**Comment by loan-laux** [comment]
I've noticed that there's no way to edit a newly added component item before clicking the Save button. Was that the case before or is this a regression?

**Comment by loan-laux** [comment]
@willopez I can't assign a reviewer here for some reason... Since you were planning on fixing this, do you want to do a code review?

**Comment by willopez** [comment]
Thanks for this @loan-laux! I'll review it, and I believe that was the case and this PR does not cause a regression. I'll verify when reviewing.

**Comment by mikemurray** [comment]
I agree with @willopez, I don’t think your changes caused a regression.

File: imports/plugins/core/navigation/client/components/NavigationDashboard.js [hunk 1]
```diff
@@ -3,7 +3,7 @@ import PropTypes from "prop-types";
 import { Dialog, DialogContent } from "@material-ui/core";
 import Button from "@reactioncommerce/catalyst/Button";
 import HTML5Backend from "react-dnd-html5-backend";
-import { DragDropContext } from "react-dnd";
+import { DndProvider } from "react-dnd";
```
Comment (mikemurray) [inline imports/plugins/core/navigation/client/components/NavigationDashboard.js]: This import seems unused. Can it be removed?

Comment (loan-laux) [reply imports/plugins/core/navigation/client/components/NavigationDashboard.js]: Good catch! A leftover from my debugging process... Thanks for that!

**Review by mikemurray — APPROVED** [review]
👍 Look good.

Had to run tests locally since they didn't run on CI. This was fixed on CircleCI and they should run on future pull-requests from forked repos.

**Review by willopez — APPROVED** [review]
LGTM

**Comment by loan-laux** [comment]
Thanks @willopez @mikemurray. Can we get a release for that? There's also a bunch of other things that are on `trunk` but not on `beta.6`, so it would make sense to release a `beta.7` IMO.

**Comment by willopez** [comment]
@loan-laux You are welcomed and thank you for you contribution. We'll get a new beta out soon.

**Pull request merged**